### DISCUSSION TITLE
Support of Avid Op-Atom files by BMX for reading and patching

### DISCUSSION THREAD

**Discussion by extenceIn** [bbc/bmx#2] [Q&A] [answered]
Hi, we are using AMT3 to create Avid OP-Atom media files to use them in Avid environment, but we also need to rewrap them into MXF Op1a to create stitched files for some internal processes. Unfortunately, we are having issues:
```
ERROR (libMXF): 'mxf_get_item(set, itemKey, &item)' failed, in ..\..\..\mxf\mxf_header_metadata.c:2250
ERROR (libMXF): 'mxf_get_uint32_item(_cMetadataSet, itemKey, &result)' failed, at ..\..\..\libMXF++\MetadataSet.cpp:180
ERROR: Failed to open MXF file '.\outputV01.mxf': general error
```
Can BMX handle the media created using AMT 3rd version? 
Here's an example of such a media file: [outputV01.zip](https://github.com/bbc/bmx/files/10013241/outputV01.zip)

**Comment by philipnbbc** [comment, marked as answer]
I had a quick look and the issue is that the file is missing the required Track Number property in the Material Package tracks. The workaround was trivial and I've pushed the commit to libMXFpp, https://github.com/bbc/libMXFpp/commit/5a27da38823569ab2fbd3009b4e14ac2dc57af17

**Reply by extenceIn** [reply]
Thank you so much, Philip. I rebuilt the apps with the fix and got it working.
